burn-core/src/module/param/primitive.rs

```diff
@@ -197,3 +197,66 @@ where
         self.map(|module| module.valid())
     }
 }
+
+macro_rules! impl_module_tuple {
+    ([$($l:ident),*][$($i:tt),*]) => {
```

Comment:
I would add a simple comment explaining this macro: like `$l` is generic modules and `$i` is the positions.

Reply:
Done!